Deprecation of a part of matplotlib in scatter

Error message is something like:

```python
=================================== FAILURES ===================================
__________________ TestPylab.test_multigraph_edgelist_tuples ___________________
self = <networkx.drawing.tests.test_pylab.TestPylab object at 0x7fe5ea4b9ba8>
    def test_multigraph_edgelist_tuples(self):
        # See Issue #3295
        G = nx.path_graph(3, create_using=nx.MultiDiGraph)
        nx.draw_networkx(G, edgelist=[(0, 1, 0)])
>       nx.draw_networkx(G, edgelist=[(0, 1, 0)], node_size=[10, 20])
drawing/tests/test_pylab.py:175: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
drawing/nx_pylab.py:273: in draw_networkx
    node_collection = draw_networkx_nodes(G, pos, **kwds)
drawing/nx_pylab.py:416: in draw_networkx_nodes
    label=label)
../matplotlib/__init__.py:1538: in inner
    return func(ax, *map(sanitize_sequence, args), **kwargs)
../matplotlib/cbook/deprecation.py:356: in wrapper
    return func(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <matplotlib.axes._subplots.AxesSubplot object at 0x7fe5ea480da0>
x = masked_array(data=[-0.3801248 , -0.23547484,  0.61559965],
             mask=False,
       fill_value=1e+20)
y = masked_array(data=[-1.        ,  0.06022964,  0.93977036],
             mask=False,
       fill_value=1e+20)
s = masked_array(data=[10, 20],
             mask=False,
       fill_value=999999)
c = '#1f78b4', marker = 'o', cmap = None, norm = None, vmin = None, vmax = None
alpha = None, linewidths = None, verts = None, edgecolors = None
plotnonfinite = False, kwargs = {'label': None}
        @_preprocess_data(replace_names=["x", "y", "s", "linewidths",
                                         "edgecolors", "c", "facecolor",
                                         "facecolors", "color"],
                          label_namer="y")
        @cbook._delete_parameter("3.2", "verts")
        def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
                    vmin=None, vmax=None, alpha=None, linewidths=None,
                    verts=None, edgecolors=None, *, plotnonfinite=False,
                    **kwargs):
```

--- a/networkx/drawing/tests/test_pylab.py
+++ b/networkx/drawing/tests/test_pylab.py
@@ -172,7 +172,7 @@ class TestPylab(object):
         # See Issue #3295
         G = nx.path_graph(3, create_using=nx.MultiDiGraph)
         nx.draw_networkx(G, edgelist=[(0, 1, 0)])
-        nx.draw_networkx(G, edgelist=[(0, 1, 0)], node_size=[10, 20])
+        nx.draw_networkx(G, edgelist=[(0, 1, 0)], node_size=[10, 20, 0])
 
     def test_alpha_iter(self):
         pos = nx.random_layout(self.G)

--- a/requirements/extras.txt
+++ b/requirements/extras.txt
@@ -1,7 +1,7 @@
 numpy>=1.15.4
 scipy>=1.1.0
 pandas>=0.23.3
-matplotlib>=3.0.2
+matplotlib>=3.1
 pygraphviz>=1.5
 pydot>=1.2.4
 pyyaml>=5.1

